tests/cli/test_remote.py

```diff
@@ -1,7 +1,731 @@
+
+
+@dont_run_for_real
+@disable_internet_access
+def test_get(mock_connection: Connection, host: str):
+    # TODO: Make this test smarter? or no need? (because we'd be testing fabric at that
+    # point?)
+    r = Remote(host, connection=mock_connection)
+    _result = r.get("foo", "bar")
+    mock_connection.get.assert_called_once_with("foo", "bar")
+
+
+@dont_run_for_real
+@disable_internet_access
+def test_put(mock_connection: Connection, host: str):
+    r = Remote(host, connection=mock_connection)
+    _result = r.put("foo", "bar")
+    mock_connection.put.assert_called_once_with("foo", "bar")
+
+
+@dont_run_for_real
+@disable_internet_access
+def test_puttext(mock_connection: Connection, host: str):
+    r = Remote(host, connection=mock_connection)
+    dest_dir = "bar/baz"
+    dest = f"{dest_dir}/bob.txt"
+    _result = r.puttext("foo", dest)
+    mock_connection.run.assert_called_once_with(
+        f"mkdir -p {dest_dir}",
+        asynchronous=False,
+        hide=True,
+        warn=False,
+        out_stream=None,
+    )
+    mock_connection.put.assert_called_once_with(unittest.mock.ANY, dest)
+
+
+@can_run_for_real
+def test_home(mock_connection: Connection, host: str, internet_enabled: bool):
+    r = Remote(host, connection=mock_connection)
+    home_dir = r.home()
+    if internet_enabled:
+        assert home_dir.startswith("/home/mila/")
+    else:
+        mock_connection.run.assert_called_once_with(
+            "echo $HOME", asynchronous=False, hide=True, warn=False, out_stream=None
+        )
+        mock_connection.local.assert_not_called()
+
+
+@pytest.mark.skipif(
+    "-s" not in sys.argv, reason="TODO: Seems to require the -s option?!"
+)
+@dont_run_for_real
+@disable_internet_access
+def test_persist(mock_connection: Connection, host: str, capsys: pytest.CaptureFixture):
+    r = Remote(host, connection=mock_connection)
+    r.persist()
+    assert (
+        "Warning: --persist does not work with --node or --job"
+        in capsys.readouterr().out
+    )
+
+
+@pytest.fixture
+def remote(mock_connection: Connection, host: str) -> Remote:
+    return Remote(hostname=host, connection=mock_connection)
+
+
+@dont_run_for_real
+@disable_internet_access
+def test_ensure_allocation(remote: Remote):
+    assert remote.ensure_allocation() == ({"node_name": remote.hostname}, None)
+
+
+def some_transform(x: str) -> str:
+    return f"echo Hello && {x}"
+
+
+def some_other_transform(x: str) -> str:
+    return f"echo 'this is printed after the command' && {x}"
+
+
+class TestSlurmRemote:
+    @dont_run_for_real
+    @pytest.mark.parametrize("persist", [True, False])
+    def test_init_(self, mock_connection: Connection, persist: bool):
+        alloc = ["--time=00:01:00"]
+        transforms = [some_transform]
+        remote = SlurmRemote(
+            mock_connection, alloc=alloc, transforms=transforms, persist=persist
+        )
+        # TODO: This kind of test feels a bit dumb.
```

Comment:
We probably only need tests that check the actual string of a command (persist or not) with a `FileRegressionFixture` or something like that